File: Amazon_LeetCode_Python_C++/ArrayString/167_two_sum_II_input_array_is_sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        vector<int> ans;
        unordered_map<int, int> mp;
        for (int i = 0; i < numbers.size(); i++) {
            if (mp.find(target - numbers[i]) != mp.end()) {
                ans.push_back(mp[target-numbers[i]]);
                ans.push_back(i + 1);
            }
            mp[numbers[i]] = i + 1;
        }
        return ans;
    }
};

File: Amazon_LeetCode_Python_C++/ArrayString/167_two_sum_II_input_array_is_sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        ans = list()
        mp = dict()
        for i, n in enumerate(numbers):
            if target - n in mp:
                ans.append(mp[target - n])
                ans.append(i + 1)
            mp[n] = i + 1
        return ans
